src/mpi/coll/allgather/allgather_inter_local_gather_remote_bcast.c

```diff
@@ -21,7 +21,7 @@ int MPIR_Allgather_inter_local_gather_remote_bcast(const void *sendbuf, int send
 {
     int rank, local_size, remote_size, mpi_errno = MPI_SUCCESS, root;
     int mpi_errno_ret = MPI_SUCCESS;
-    MPI_Aint sendtype_sz;
+    MPI_Aint sendtype_sz = 0;
```

Comment:
Fixing like this just blindly suppresses the warning -- equivalent to disable that warning option. Preferably we should find the actual branch that uses the uninitialized variable and fix that branch only.